relations/src/lib.rs

```diff
@@ -18,6 +19,7 @@ pub use environment::{
 };
 pub use linear::LinearEquationRelation;
 pub use merkle_tree::{MerkleTreeRelation, Root};
+#[cfg(any(feature = "preimage", feature = "preimage-std"))]
```

Comment:
I understand that it's so that the project compiles but in which situations you imagine we would enable these features?

Also, if we only ever use them in the combination like this - _any of the two_ - can we replace it with a single feature?

Reply:
> in which situations you imagine we would enable these features

once we substitute 'tangling' with 'poseidon'

> replace it with a single feature

if 'poseidon' is compiling perfectly as we intend, there'll be no feature; otherwise, we might need it in both `std` and `no-std` context